### PR TITLE
Benji/ios 282 update conversation header

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: iOS/Collection View Management /CollectionView.swift [hunk 1]
```diff
@@ -24,7 +24,10 @@ class CollectionView: UICollectionView {
     }
 
     func initializeViews() {
-        self.addSubview(self.animationView)
+        let backView = View()
```
Comment (Benjamindodgson) [inline iOS/Collection View Management /CollectionView.swift:27]: The animation view was centering correctly. This fixes that.

File: iOS/Collection View Management /DiffableCollectionViewController.swift [hunk 2]
```diff
@@ -59,6 +59,7 @@ class DiffableCollectionViewController<SectionType: Hashable, ItemType: Hashable
 
     @MainActor
     func loadData() async {
+        self.collectionView.layoutNow()
```
Comment (Benjamindodgson) [inline iOS/Collection View Management /DiffableCollectionViewController.swift:62]: Helps with centering.

File: iOS/Flows/Conversation/Conversation Header/ConversationHeaderViewController.swift [hunk 3]
```diff
@@ -12,39 +12,28 @@ import Combine
 import Lottie
 import UIKit
 
-class ConversationHeaderView: View {
+class ConversationHeaderViewController: ViewController, ActiveConversationable {
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Conversation Header/ConversationHeaderViewController.swift:15]: Made this a controller

File: iOS/Flows/Conversation/Conversation Header/ConversationHeaderViewController.swift [hunk 4]
```diff
@@ -65,46 +54,32 @@ class ConversationHeaderView: View {
...
-        let members = conversation.lastActiveMembers.filter { member in
-            return member.id != ChatClient.shared.currentUserId
-        }
-
-        if !members.isEmpty {
-            self.stackedAvatarView.set(items: members)
-        } else {
-            self.stackedAvatarView.set(items: [User.current()!])
-        }
+        ConversationsManager.shared.$activeConversation
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Conversation Header/ConversationHeaderViewController.swift:58]: How to access the active conversation updates.

File: iOS/Flows/Conversation/Conversation Header/MemberCell.swift [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+//
+
+import Foundation
+import StreamChat
+
+class MemberCell: CollectionViewManagerCell, ManageableCell {
+
+    typealias ItemType = AnyHashableDisplayable
+
+    var currentItem: AnyHashableDisplayable?
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Conversation Header/MemberCell.swift:16]: I forgot we had this object!

File: iOS/Flows/Conversation/Conversation Header/MembersViewController.swift [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        let members = conversation.lastActiveMembers.filter { member in
+            return member.id != ChatClient.shared.currentUserId
+        }
+
+        if !members.isEmpty {
+            data[.members] = members.compactMap({ member in
+                return .member(AnyHashableDisplayable.init(member))
+            })
+        } else {
+            data[.members] = [.member(AnyHashableDisplayable.init(User.current()!))]
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/Conversation Header/MembersViewController.swift:64]: This is nice cause we can add either a user or member.

File: iOS/Flows/Conversation/ConversationListCoordinator.swift [hunk 7]
```diff
@@ -13,7 +13,7 @@ import PhotosUI
 import Combine
 import StreamChat
 
-class ConversationListCoordinator: PresentableCoordinator<Void> {
+class ConversationListCoordinator: PresentableCoordinator<Void>, ActiveConversationable {
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/ConversationListCoordinator.swift:16]: This protocol was used in quite a few places.

**Review by MartinJibber — APPROVED** [review]

**Pull request merged**